conclave-host/src/main/kotlin/com/r3/conclave/host/internal/EnclaveHandle.kt

```diff
@@ -25,6 +25,15 @@ interface EnclaveHandle {
      */
     val mockEnclave: Any
 
+    /**
+     * This is to establish which entity triggers the quoting: HOST or ENCLAVE.
+     * In Graal/Native Image it is responsibility of the host to use sgx_qe_get_quote and
+     * pass the result of it to the enclave.
+     * In Gramine it is responsibility of the enclave (and Gramine internals) to get the quote through AESM service.
+     * Note: Mock mode is following Graal/Native Image approach: the host does a call to the enclave to get the quote.
+     */
+    val quotingManager: QuotingManager
```

Comment:
Again, don't fully understand this.

Reply:
I think the one reason you've introduced this enum is bc mock mode emulates GraalVM. But I don't believe that's correct. Let it do what makes sense for it. If Gramine and GraalVM behaved the same then yes, it would make sense for mock to do the same as well. But since that's not the case, let's not force GraalVM behaviour on mock mode.

Reply:
I have introduced this to avoid the ugly `is NativeHandle` vs `is GramineHandle`, to distinguish between the flow to get the "signed quote" when working in Graal vs Gramine.

Mock mode is somehow untouched by my changes, it is doing what was doing before, i.e. following the same approach as Graal.